cron and cronvar - Use get_bin_path() to find executable

JeanFred opened the pull request:
##### SUMMARY

Instead of systematically using `/usr/bin/crontab`, this uses the `get_bin_path` utility to locate the default `crontab` executable used in the cron/cronvar tasks.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
cron
cronvar

##### ADDITIONAL INFORMATION

In some remote environments, the `crontab` executable is
overloaded with a custom executable, which typically does
some pre/post processing before forwarding to crontab.

JeanFred commented:
Regarding the Ansible version: was not too sure, please advise :-)

Regarding unit tests: these modules do not appear to have any unit tests.

JeanFred commented:
Is there anything else I need to do to help move this forward?

JeanFred commented:
(Rebased on the latest `devel` to unstale the CI.)

JeanFred commented:
(Rebased, again, on the latest devel to unstale the CI.)

JeanFred commented:
Changed the version to 2.10 and rebased on latest `devel`.

samdoran commented on lib/ansible/modules/system/cron.py:
```diff
@@ -216,7 +228,7 @@
 
 
-CRONCMD = "/usr/bin/crontab"
+DEFAULT_CRONCMD = "/usr/bin/crontab"
```
Rather than adding a parameter to override the hard coded path to the executable, use `get_bin_path()` from `lib/ansible/module_utils/common/process.py`. Changing to using that function may remove the need for a parameter for the executable.

If you wanted to add a parameter to the module, it could be an executable name or a list of additional paths to search (or both, though that's probably overkill).

JeanFred replied:
I was not aware of `get_bin_path`! Yep, that would solve my use case.

JeanFred committed:
Use the default crontab executable in cron and cronvar modules

In some remote environments, the `crontab` executable is
overloaded with a custom executable, which typically does
some pre/post processing before forwarding to crontab.

Instead of using the hardcoded `/usr/bin/crontab`, this uses
the `get_bin_path` utility to locate the default crontab executable.

JeanFred commented:
Following @samdoran advice, this is now simplified and just using `get_bin_path`. I confirmed this works with the environment I need that for (by having the forked version in `library`)

samdoran commented:
Very nice. This is a much simpler fix. Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

I think we should also reclassify this as a bugfix since the module fails if the executable isn't located at a specific path.

JeanFred commented:
> Very nice. This is a much simpler fix. Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/stable-2.7/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Added a changelog fragment.
